atoma-proxy/src/server/handlers/nodes.rs

```diff
@@ -285,6 +288,14 @@ pub(crate) async fn nodes_models_retrieve(
             stack_small_id: stack_small_id as u64,
         }))
     } else {
+        // NOTE: We need to check the user's balance before acquiring a new stack entry.
```

Comment:
I don't understand this comment. You can't check balance without auth, because you don't know the user id.